### PR TITLE
[ARCTIC-1745] Only skip partitions when the maximum size limit is exceeded

### LINKED ISSUE #1745: [Bug]: The limit of the number of partitions processed in one Optimizing process doesn't work
### What happened?

According to the design, an optimizing process should not handle too much data at once to avoid resources being monopolized by a single optimizing process for a table. By default, this limit is set to 500MB. 

Under this limit, for partitioned tables, an optimizing process is allowed to process only a subset of partitions at a time. 

However, it was found during testing that an optimizing process processed multiple partitions and the data volume far exceeded 500MB, like

image

### Affects Versions

0.5.0

### What engines are you seeing the problem on?

_No response_

### How to reproduce

1.create an partitioned iceberg format table
2.write to 3 partitions of the table, and write more than 500M data files to each partition
3.enable the optimizing of table
4.we could find an optimizing process handling data files for all partitions

### Relevant log output

_No response_

### Anything else

Bug in the calculation of actualInputSize.
image

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

fix #1745 , if the first partition's size > maxSize, it should not be skipped

## Brief change log

  - fix skip all partitions when the first partition's size > maxSize

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
 public class OptimizingPlanner extends OptimizingEvaluator {
   private static final Logger LOG = LoggerFactory.getLogger(OptimizingPlanner.class);
 
-  private static final long MAX_INPUT_FILE_SIZE_PER_THREAD = 5L * 1024 * 1024 * 1024;
+  private static final long MAX_INPUT_FILE_SIZE_PER_THREAD = 512 * 1024 * 1024; // 500MB
```
Comment (XBaith) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:43]: 512MB?

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java [hunk 2]
```diff
@@ -288,8 +289,8 @@ private void planTasks() {
     }
   }
 
-  private double getAvailableCore(TableRuntime tableRuntime) {
-    return tableRuntime.getOptimizingConfig().getTargetQuota();
+  private double getAvailableCore() {
+    return authOptimizers.values().stream().map(Resource::getThreadCount).reduce(Integer::sum).orElse(0);
```
Comment (zhongqishang) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java:293]: Maybe we should match table group ?

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java:293]: Do you mean the `Optimizer Group`? All the `authOptimizers` have the same `Optimizer Group` in the object of `OptimizingQueue`.

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java (hunk 1 above)
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:43]: Yes, I will fix it.

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java (hunk 2 above)
Comment (zhongqishang) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java:293]: Yes, you are right. My fault.

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java [hunk 3]
```diff
@@ -123,11 +123,11 @@ public List<TaskDescriptor> planTasks() {
     List<PartitionEvaluator> inputPartitions = Lists.newArrayList();
```
Comment (XBaith) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:123]:
```
    double maxInputSize = MAX_INPUT_FILE_SIZE_PER_THREAD * availableCore;
```
`availableCore` should not be zero

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:123]: Okay, I think setting the minimum value of availableCore to 1 is reasonable.

Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:123]: nerver mind, 0 `maxInputSize` doesn't really affect anything for planing

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**